src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/analytics/custom.py

```diff
@@ -0,0 +1,229 @@
+                                                                       new_credential_user_password,
+                                                                       uri,
+                                                                       credential_user_name)
+    client.update_credential(account_name, database_name, credential_name, update_params)
+
+# job customizations
+# pylint: disable=too-many-arguments
+def submit_adla_job(client,
+                    account_name,
+                    job_name,
+                    script,
+                    runtime_version=None,
+                    compile_mode=None,
+                    compile_only=False,
+                    degree_of_parallelism=1,
+                    priority=1000):
+    contents = None
+    # script can be either script contents or a path to a script
+    # we attempt to always open the script path first, if that fails
+    # then treat it as contents
+    try:
+        with open(script, 'r') as f:
+            contents = f.read()
+    #pylint: disable=broad-except
+    except Exception:
+        contents = script
+
+    if not contents:
+        # pylint: disable=line-too-long
+        raise CLIError('Could not read script content from the supplied --script param. It is either empty or an invalid file. value: {}'.format(script))
+
+    job_properties = USqlJobProperties(contents)
+    if runtime_version:
+        job_properties.runtime_version = runtime_version
+
+    if compile_mode:
+        job_properties.compile_mode = compile_mode
+
+    submit_params = JobInformation(job_name,
+                                   JobType.usql,
+                                   job_properties,
+                                   degree_of_parallelism,
+                                   priority)
+    if compile_only:
+        return client.build(account_name, submit_params)
+
+    job_id = _get_uuid_str()
+
+    return client.create(account_name, job_id, submit_params)
+
+# pylint: disable=superfluous-parens
+def wait_adla_job(client,
+                  account_name,
+                  job_id,
+                  wait_interval_sec=5,
+                  max_wait_time_sec=-1):
+    if wait_interval_sec < 1:
+        # pylint: disable=line-too-long
+        logger.warning('wait times must be positive when polling jobs, setting to the default value of 5 seconds')
+        wait_interval_sec = 5
+
+    job = client.get(account_name, job_id)
+    time_waited_sec = 0
+    while job.state != JobState.ended:
+        if max_wait_time_sec > 0 and time_waited_sec >= max_wait_time_sec:
+            # pylint: disable=line-too-long
+            raise CLIError('Data Lake Analytics Job with ID: {0} has not completed in {1} seconds. Check job runtime or increase the value of --max-wait-time-sec'.format(job_id, time_waited_sec))
+        print('Job is not yet done. Current job state: \'{0}\''.format(job.state))
```

Comment:
Recommend logging this as info so it only appears with --verbose (or debug). If you want it to print normally, then it should be printed to STDERR, not STDOUT.